src/thor/costmatrix.cc

```diff
@@ -321,7 +344,7 @@ void CostMatrix::ForwardSearch(const uint32_t index, const uint32_t n,
 
   // Get edge label and check cost threshold
   EdgeLabel pred = edgelabels[pred_idx];
-  if (pred.cost().secs > cost_threshold_) {
+  if (pred.cost().secs > GetCostThreshold()) {
```

Comment:
shouldn't this be set once? Don't like the idea of a switch statement being evaluated on each iteration.